test/integration/imagechange_buildtrigger_test.go

```diff
@@ -42,17 +42,17 @@ func TestSimpleImageChangeBuildTrigger(t *testing.T) {
 		t.Fatalf("Couldn't create ImageRepository: %v", err)
 	}
 
-	created, err := openshift.Client.BuildConfigs(testNamespace).Create(config)
-	if err != nil {
-		t.Fatalf("Couldn't create BuildConfig: %v", err)
-	}
-
-	watch2, err := openshift.Client.BuildConfigs(testNamespace).Watch(labels.Everything(), labels.Everything(), created.ResourceVersion)
+	watch2, err := openshift.Client.BuildConfigs(testNamespace).Watch(labels.Everything(), labels.Everything(), "0")
 	if err != nil {
 		t.Fatalf("Couldn't subscribe to BuildConfigs %v", err)
 	}
 	defer watch2.Stop()
 
+	if _, err := openshift.Client.BuildConfigs(testNamespace).Create(config); err != nil {
```

Comment:
I was getting sporadic Added events on the watch channel with the previous ordering. Didn't dig deeper. That explains the dual-watches. I don't see any explanation as to why there should be an Added AND a Modified event on that channel when the watch was established AFTER the Create and from the created resource's version. Something is going on here.